### PR TITLE
Added new subworkflow 'ANNOTATION_STATS' and new parameter --annotation_set to solve issue #121

### PR DESCRIPTION
**Pull request by SandraBabirye**
### Changes
- Added `--annotation_set` parameter to the nextflow.confiig, and main workflow.
- Updated the modules.json file  with information for two nf-core modules (agat/spstatistics and agat/sqstatbasic).
- Modified the `test.config and test_full.config` file to include path to the test dataset that for annotation file specified when  athe `--annotation_set` parameter is defined
- Updated the nextflow_schema.json file with details and usage for params.annotation_set

### CODE REVIEW

**Comment by SandraBabirye** [comment]
Hi @BethYates  and @tkchafin , I am kindly requesting for your review on this PR to solve issue #121 .

File: nextflow_schema.json
Comment (BethYates) [inline nextflow_schema.json:1]: The "pattern"  on line 148 should use gff rather than gff3

File: subworkflows/local/annotation_statistics.nf
Comment (BethYates) [inline subworkflows/local/annotation_statistics.nf:1]: line 4 has the wrong path, it should be `include { AGAT_SQSTATBASIC  } from '../../modules/nf-core/agat/sqstatbasic/main'`

File: workflows/genomenote.nf
Comment (BethYates) [inline workflows/genomenote.nf:1]: You need to include the ANNOTATION_STATS subworkflow by adding  `include { ANNOTATION_STATS  } from '../subworkflows/local/annotation_statistics' ` where the other subworkflows are added

File: conf/base.config
Comment (BethYates) [inline conf/base.config:1]: When running the test profile AGAT_SPSTATISTICS runs out of memory, can you try running the pipeline with the test profile to work out what the memory request for this process should be increased to?

**Review by BethYates — APPROVED** [review]
Good work @SandraBabirye!

**Pull request merged**